openstack_hypervisor/hooks.py

```diff
@@ -164,6 +182,11 @@ def install(snap: Snap) -> None:
     """
     setup_logging(snap.paths.common / "hooks.log")
     logging.info("Running install hook")
+
+    for option, default in DEFAULT_CONFIG.items():
+        if callable(default):
+            default = default()
+            DEFAULT_CONFIG[option] = default
     logging.info(f"Setting default config: {DEFAULT_CONFIG}")
     snap.config.set(DEFAULT_CONFIG)
```

Comment:
I think we could possible switch this to use the _update_default_config function